### PR TITLE
Added inline buttons

### PR DESCRIPTION
**Pull request by athulkrishnaaei**
Added inline buttons for setting template,global template,get filter,get template,unsubscribe,remove template etc

### CODE REVIEW

File: src/http_client.rs [hunk 1]
```diff
@@ -1,26 +0,0 @@
-use crate::config::Config;
```
Comment (ayrat555) [inline src/http_client.rs:1]: let's return this

File: src/sync/reader.rs [hunk 2]
```diff
@@ -48,14 +49,21 @@ pub trait ReadFeed {
 }
 
 pub fn read_url(url: &str) -> Result<Vec<u8>, FeedReaderError> {
-    let client = http_client::client();
-
-    let request = Request::get(url)
-        .header("User-Agent", "el_monitorro")
+    let client = match Request::get(url)
```
Comment (ayrat555) [inline src/sync/reader.rs:52]: let's use changes from the master branch

File: src/sync/sync_feed_job.rs [hunk 3]
```diff
@@ -96,7 +96,7 @@ impl SyncFeedJob {
 
         let message = format!("{} can not be processed. It was removed.", feed.link);
 
-        let api = telegram_client::api();
+        let api = Api::default();
```
Comment (ayrat555) [inline src/sync/sync_feed_job.rs:99]: please revert

**Review by ayrat555 — COMMENTED** [review]
- Very good but has some room for improvement
- Let's use changes from the master branch
- Let's continue improving this pr (do not close and open again like you like to do)

File: src/bot/commands.rs [hunk 4]
```diff
@@ -30,12 +39,25 @@ pub mod set_content_fields;
...
+pub mod set_template_inline_keyboard;
 pub mod set_timezone;
 pub mod start;
 pub mod subscribe;
+pub mod subscribe_inline_keyboard;
 pub mod unknown_command;
 pub mod unsubscribe;
+pub mod unsubscribe_inline_keyboard;
+
+enum IncomingCommand {
```
[the remark this answers is not in the document]
Comment (athulkrishnaaei) [reply src/bot/commands.rs:53]: right i have only made enum for the commands that show inline keyboards

File: src/bot/handler.rs [hunk 5]
```diff
@@ -137,4 +255,256 @@ impl Handler {
     fn owner_telegram_id() -> Option<i64> {
         Config::owner_telegram_id()
     }
+
+    fn process_callback_query(
```
Comment (pxp9) [inline src/bot/handler.rs:259]: Result here and manage unwraps with `?` operator please.

**Review by pxp9 — CHANGES_REQUESTED** [review]
Keep the good work @athulkrishnaaei , just a few comments.

File: src/bot/handler.rs [hunk 6]
```diff
@@ -137,4 +255,256 @@ impl Handler {
...
+                api.delete_message(&delete_message_params).unwrap();
+                let send_message_params =
+                    SetGlobalTemplateInlineKeyboard::set_global_template_keyboard(message);
+                api.send_message(&send_message_params).unwrap();
+            }
+            _ => UnknownCommand::execute(db_pool, api, message),
+        }
+    }
+
+    fn parse_int_from_string(command: &str) -> Option<std::string::String> {
```
[the remark this answers is not in the document]
Comment (athulkrishnaaei) [reply src/bot/handler.rs:474]: To filter out the feed id number from callback data

File: src/bot/commands.rs [hunk 7]
```diff
@@ -65,16 +103,78 @@ pub trait Command {
...
+        let delete_message_params = DeleteMessageParams::builder()
+            .chat_id(message.chat.id)
+            .message_id(message.message_id)
+            .build();
 
-        self.reply_to_message(api, message, text)
+        match command {
+            Commands::Subscribe => {
+                let send_message_params = SubscribeInlineKeyboard::set_subscribe_keyboard(message);
+                api.send_message(&send_message_params).unwrap();
```
[the remark this answers is not in the document]
Comment (athulkrishnaaei) [reply src/bot/commands.rs:141]: According to frankestien documentation https://github.com/ayrat555/frankenstein/blob/master/examples/inline_keyboard.rs
unwrap is used

File: src/bot/commands.rs [hunk 8]
```diff
@@ -65,16 +103,78 @@ pub trait Command {
...
+        let feeds = data.split('\n');
+        let feeds_ids = feed_id.split(',').clone();
+        let text = self.response(db_pool.clone(), &message, &api);
+        let delete_message_params = DeleteMessageParams::builder()
+            .chat_id(message.chat.id)
+            .message_id(message.message_id)
+            .build();
 
-        self.reply_to_message(api, message, text)
+        match command {
```
Comment (ayrat555) [inline src/bot/commands.rs:138]: this is very hacky way to intercept the command. let's modify `list_subscriptions` instead

**Comment by ayrat555** [comment]
@athulkrishnaaei :crying_cat_face:

**Pull request closed without merging**